### ISSUE TITLE
Emulation - Panel's internal container does not default to Panel's size

### ISSUE THREAD

**Issue by yishayw** [apache/royale-asjs#1208]
In below example you can see in Royale the content expanding beyond the panel's height, whereas in Flex there's a scrollbar in the Panel.

```
<?xml version="1.0"?>
<!-- dragdrop\DandDCanvas.mxml -->
<mx:Application xmlns:fx="http://ns.adobe.com/mxml/2009"
	xmlns:s="library://ns.apache.org/royale/spark"
	xmlns:mx="library://ns.apache.org/royale/mx"
        width="100%" height="100%"
	backgroundColor="white">

<fx:Script>
	<![CDATA[
		]]>
</fx:Script>
        <mx:Panel width="111" height="111">
                <mx:Canvas width="100%" height="200" backgroundColor="red"/>
        </mx:Panel>
</mx:Application>
```

**Comment by greg-dove** [comment]
The problem here is the minHeight getter on PanelInternalContainer which is defaulting to measuredMinHeight.
Simply setting 
this.minHeight = 0;
in the PanelInternalContainer constructor might be one option to fix this. The way this is set up is quite different to how flex manages the content pane I think, so we might just have to apply some changes to get the behavior to be the same or at least similar. I did not check this in Flex.

**Comment by yishayw** [comment]
This is in contrast to the following scenario, which works as expected:

```
<?xml version="1.0"?>
<mx:Application xmlns:fx="http://ns.adobe.com/mxml/2009"
	xmlns:s="library://ns.apache.org/royale/spark"
	xmlns:mx="library://ns.apache.org/royale/mx"
        width="100%" height="100%"
	backgroundColor="white">

<fx:Script>
	<![CDATA[
		]]>
</fx:Script>
        <mx:VBox width="111" height="111">
                <mx:Canvas id="panelTitleBar" width="100%" height="20" backgroundColor="blue"/>
                <mx:Canvas id="panelContentContainer" width="100%" height="100%">
                        <mx:Canvas width="100%" height="200" backgroundColor="red"/>
                </mx:Canvas>
        </mx:VBox>
</mx:Application>
```

**Comment by yishayw** [comment]
This problem may have been introduced in 8af26f74e1e567ac47b936937a4072f2c2b19a53 by @aharui 

Specifically, this line
```

                if (!isNaN(percentHeight))
                    _measuredMinHeight = mh;
```
in UIComponent.get__measuredHeight() sets measured min height of panel's internal container to its content measured height. This essentially overrides BoxLayout's flexing algorithm for percentHeight children. 

As mentioned above layout is ok with other components, so I did not want to change BoxLayout without understanding this better.

I solved this by creating a custom Panel layout for mx, same as there exists one for spark. Panel is essentially just a titlebar and an internal content container (ignoring unimplemented control bar for now), so a simpler layout makes more sense to me.
